### PR TITLE
add property is_border_required

### LINKED ISSUE #3: Remove the yellow border with  `GraphicsCaptureSession.IsBorderRequired`
Is it possible to remove the yellow border when capturing?

Here is the [GraphicsCaptureSession.IsBorderRequired](https://learn.microsoft.com/en-us/uwp/api/windows.graphics.capture.graphicscapturesession.isborderrequired?view=winrt-22621) API reference.

### PR DESCRIPTION
**Pull request by Elypha**
hi,

thanks for your project

I hope this closes https://github.com/modelflat/zbl/issues/3

### CODE REVIEW

**Comment by Elypha** [comment]
also if I may suggest to rename the other property `capture_cursor` to `is_cursor_capture_enabled`,
since if this project will remain largely a wrapper/extension around the `Windows.Graphics.Capture` API, it might be better to just keep their names from the original API

it's longer, yes, but I assume it's less likely your users write code on notepad so they won't be angry with the change x

**Comment by modelflat** [comment]
@Elypha Hi, thanks for contributing! Couple of notes:
- I'm on board with renaming `capture_cursor` - good call!
- Examples won't compile after the change - could you please adjust them accordingly? (ideally we should move to use builder pattern to avoid such breaking changes in the future, but let's leave this for another time for now)
- The change doesn't really work for me. I'm on Windows 10 and am getting:
```
failed to initialize capture: Error { code: HRESULT(0x80004002), message: "No such interface supported" }
```
irrespective of whether I set `is_border_required` to true or false. Not sure what the root cause is. Does it work for you?

**Comment by Elypha** [comment]
@modelflat thanks for the feedback!

I'm trying to fix the examples in [`98654ce` (#4)](https://github.com/modelflat/zbl/pull/4/commits/98654cedd07ca2a8ec4e02f6900a40f3101e49c3) and hope they are working

For the capture init error, it seems like args are not passed to the constructor properly but I'm not very familiar with rust so I'm likely mistaken.
I'm using the Python lib and it works great. My machine runs Windows 11 22H2 and Python 3.12, and here's a dev build if you are interested to test it x
[dist.zip](https://github.com/modelflat/zbl/files/15364814/dist.zip)

**Comment by Elypha** [comment]
wait I asked mr gpt and the error code indicates something related to the os api
then I checked the MS api manual, and yes, the border option is not supported in some versions in windows 10: 

Device family | Windows 10, version 2104 (introduced in 10.0.20348.0)

https://learn.microsoft.com/en-gb/uwp/api/windows.graphics.capture.graphicscapturesession.isborderrequired?view=winrt-20348

File: zbl/src/capture/mod.rs [hunk 1]
```diff
@@ -68,7 +69,8 @@ impl Capture {
         )?;
 
         let session = frame_pool.CreateCaptureSession(&capture_item)?;
-        session.SetIsCursorCaptureEnabled(capture_cursor)?;
+        session.SetIsCursorCaptureEnabled(is_cursor_capture_enabled)?;
+        session.SetIsBorderRequired(is_border_required)?;
```
Comment (modelflat) [inline zbl/src/capture/mod.rs]: Let's account for the possibility of this not being supported with a meaningful error message. Something like this should work:

```suggestion
        if !is_border_required {
            if let Err(e) = session.SetIsBorderRequired(is_border_required) {
                log::warn!(
                    "got '{}' when trying to disable the capture border - see https://github.com/modelflat/zbl/pull/4 for more info",
                    e
                );
            }
        }
```

File: zbl_py/zbl/__init__.py [hunk 2]
```diff
@@ -71,7 +72,8 @@ def show(args):
         with Capture(
             window_name=args.window_name,
             display_id=args.display_id,
-            capture_cursor=args.capture_cursor,
+            is_cursor_capture_enabled=args.is_cursor_capture_enabled,
+            is_border_required=args.is_border_required,
```
Comment (modelflat) [inline zbl_py/zbl/__init__.py:76]: Let's also update the args in `zbl_py/zbl/__main__.py` for the CLI tool.

**Review by modelflat — CHANGES_REQUESTED** [review]
Yup, this seems to be unsupported on the Windows version I'm on. But, let's proceed with this change - just a couple of fixes and it should be ready to go!

**Comment by Elypha** [comment]
great, can you test if all the changes work now

File: zbl/examples/directx_interop.rs [hunk 3]
```diff
@@ -28,10 +28,11 @@ fn main() {
         panic!("either --window-name or --display-id should be set!");
     };
 
-    let mut capture = Capture::new(target, true, false).expect("failed to initialize capture");
-
-    opencv::core::initialize_context_from_d3d11_device(&mut capture.d3d.device)
-        .expect("initialize d3d11");
+    let mut capture = Capture::new(target, false, false, false).expect("failed to initialize capture");
+    unsafe {
+        opencv::core::initialize_context_from_d3d11_device(&capture.d3d.device)
```
Comment (modelflat) [inline zbl/examples/directx_interop.rs]:
```suggestion
        opencv::core::initialize_context_from_d3d11_device(&mut capture.d3d.device)
```

File: zbl_py/src/lib.rs [hunk 4]
```diff
@@ -122,23 +125,27 @@ impl Capture {
...
         window_handle: Option<i32>,
         display_id: Option<i32>,
-        capture_cursor: Option<bool>,
+        is_cursor_capture_enabled: Option<bool>,
+        is_border_required: Option<bool>,
         cpu_access: Option<bool>,
     ) -> PyResult<Self> {
-        let capture_cursor = capture_cursor.unwrap_or(false);
+        let is_cursor_capture_enabled = is_cursor_capture_enabled.unwrap_or(false);
+        let is_border_required = is_border_required.unwrap_or(false);
```
Comment (modelflat) [inline zbl_py/src/lib.rs]: I think we should change the default to be `true` (and in the Python version as well) to mirror the WGC API.

**Review by modelflat — CHANGES_REQUESTED** [review]
Almost there - let's just change the default value for the `IsBorderRequired` to mirror the WGC API default (`true`)

**Comment by Elypha** [comment]
sorry microsoft /j
the default values in the rust constructor and python lib are updated

**Comment by modelflat** [comment]
Merged! Thank you so much for your contribution ❤️

**Pull request merged**